spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/SecretsPropertySourceLocator.java

```diff
@@ -60,25 +61,27 @@ public SecretsPropertySourceLocator(SecretsConfigProperties properties) {
 	}
 
 	@Override
-	public PropertySource locate(Environment environment) {
+	public PropertySource<?> locate(Environment environment) {
 		if (environment instanceof ConfigurableEnvironment) {
 			ConfigurableEnvironment env = (ConfigurableEnvironment) environment;
 
 			List<SecretsConfigProperties.NormalizedSource> sources = this.properties.determineSources();
+			Set<SecretsConfigProperties.NormalizedSource> uniqueSources = new HashSet<>(sources);
+			LOG.debug("Secrets normalized sources : " + sources);
 			CompositePropertySource composite = new CompositePropertySource("composite-secrets");
 			// read for secrets mount
 			putPathConfig(composite);
 
 			if (this.properties.isEnableApi()) {
-				sources.forEach(s -> composite.addPropertySource(getKubernetesPropertySourceForSingleSecret(env, s)));
+				uniqueSources.forEach(s -> composite.addPropertySource(getMapPropertySourceForSingleSecret(env, s)));
 			}
 
 			return composite;
 		}
 		return null;
 	}
 
-	private MapPropertySource getKubernetesPropertySourceForSingleSecret(ConfigurableEnvironment environment,
```

Comment:
this is a nitpick since config map names these methods like this, so just to be on par